src/main/scala/ai/privado/utility/Utilities.scala

```diff
@@ -151,6 +164,39 @@ object Utilities {
     }
   }
 
+  def addRuleTagsForGA(
+    builder: BatchedUpdate.DiffGraphBuilder,
+    node: AstNode,
+    ruleInfo: RuleInfo,
+    ruleCache: RuleCache,
+    cpg: Cpg,
+    ruleId: Option[String] = None
+  ): Unit = {
+    // Generate Arguments for the GTM & Segment Pixels
+    if (checkIfGTMOrSegment(ruleInfo.id)) {
+      if (node.isInstanceOf[Call]) {
+        try {
+          val storeForTagHelper = storeForTag(builder, node, ruleCache) _
+          val callNode          = node.asInstanceOf[Call]
+          val argumentList: List[(String, String)] =
+            try {
+              SinkArgumentUtility.addArgumentsForGAPixelNode(callNode, cpg)
+            } catch {
+              case ex: Exception =>
+                logger.debug(s"An error occurred while processing arguments: ${ex.getMessage}")
+                List.empty[(String, String)] // Returning an empty list as a default value
```

Comment:
better to also log some information like call Node code, arguments etc for debugging purpose